state/src/replicationv2/mod.rs

```diff
@@ -12,13 +12,13 @@ mod state_machine;
 
 use openraft::{EmptyNode, Raft as RaftInner, RaftTypeConfig};
 
-use crate::StateTransition;
+use crate::Proposal;
 
 // Declare the types config for the raft
 openraft::declare_raft_types! (
     /// The type config for the raft
     pub TypeConfig:
-        D = Box<StateTransition>,
+        D = Proposal,
```

Comment:
I'm assuming we updated the `Proposal` struct to contain a `Box<StateTransition>` instead of just a `StateTransition` and that's just not in this PR?

Reply:
yeah the rest of it's file is in an intermediate refactor state so I left it out